chroma_core/services/job_scheduler/job_scheduler.py

```diff
@@ -872,8 +874,9 @@ def notify(self, content_type, object_id, time_serialized, update_attrs, from_st
     def run_jobs(self, job_dicts, message):
         with self._lock:
             result = self.CommandPlan.command_run_jobs(job_dicts, message)
-            self.progress.advance()
-            return result
+
+        self.progress.advance()
```

Comment:
Why is this moved outside the lock?

Reply:
I put it inside the lock.

Reply:
I probably moved it out of the lock initially because in other spots, such as `test_host_contact`, the `advance` was moved outside the lock.